lib/models/repository-states/present.js

```diff
@@ -440,11 +504,14 @@ export default class Present extends State {
     });
   }
 
-  getCurrentBranch() {
-    return this.cache.getOrSet(Keys.currentBranch, async () => {
-      const {name, isDetached} = await this.git().getCurrentBranch();
-      return isDetached ? Branch.createDetached(name) : new Branch(name);
-    });
+  async getCurrentBranch() {
+    const {branch} = await this.getStatusBundle();
+    if (branch.head === '(detached)') {
```

Comment:
Yeah it is super edge-casey. If we really wanted to address it we could first check to see if `(detached)` is listed in the results of `getBranches()`. But it seems like it might not be worth it to do this check every time we suspect we're in detached head state since it is such an edge case